File: 3396-valid-word/valid-word.ts
const VOWELS = ['a', 'e', 'i', 'o', 'u']
function isValid(word: string): boolean {
    if(word.length < 3) return false;
    let vowel = false, consonant = false
    for(let i = 0; i < word.length; i++) {
        const charCode = word.charCodeAt(i)
        if(charCode >= 48 && charCode <= 57) {}
        else if ((charCode >= 97 && charCode <= 122) || (charCode >= 65 && charCode <= 90)) {
            if(VOWELS.includes(word[i].toLowerCase())) vowel = true
            else consonant = true
        } else return false
    }
    
    return vowel && consonant
};

File: 3396-valid-word/valid-word.rs
impl Solution {
    pub fn is_valid(word: String) -> bool {
        if word.len() < 3 {
            return false;
        } 
        let mut vowel = 0;
        let mut conso = 0;
        for v in word.chars() {
            let is_digit = v.is_ascii_digit();
            if !v.is_alphabetic() && !is_digit {
                return false;
            }
            if "aiueoAIUEO".contains(v) {
                vowel += 1;
            } else if !is_digit {
                conso += 1;
            }
        }
        vowel >= 1 && conso >= 1
    }
}